cltk/phonology/old_swedish/transcription.py

```diff
@@ -0,0 +1,197 @@
+dh = Consonant("dental", "frictative", True, "ð", False)
+
+OLD_NORSE8_PHONOLOGY = [
+    a, ee, i, oee, y, u, o, a.lengthen(), ee.lengthen(), e.lengthen(), oee.lengthen(),
+    i.lengthen(), y.lengthen(), u.lengthen(), o.lengthen(),
+    p, b, t, d, k, g, f, v, th, dh, s, gh, h, j, l, r, n
+]
+
+# IPA Dictionary
+DIPHTHONGS_IPA = {
+    "ey": "ɐy",  # Diphthongs
+    "au": "ɒu",
+    "øy": "ɐy",
+    "ei": "ei",
+}
+# Wrong diphthongs implementation but not that bad for now
+DIPHTHONGS_IPA_class = {
+    "ey": Vowel("open", "front", True, "short", "ɐy"),
+    "au": Vowel("open", "back", True, "short", "ɒu"),
+    "øy": Vowel("open", "front", True, "short", "ɐy"),
+    "ei": Vowel("open", "front", True, "short", "ɛi"),
+}
+IPA = {
+    "a": "a",  # Short vowels
+    "e": "ɛ",
+    "i": "i",
+    "o": "o",
+    "ö": "ø",
+    "ø": "ø",
+    "u": "u",
+    "y": "y",
+    "æ": "ɛ",
+    # Consonants
+    "b": "b",
+    "c": "k",
+    "d": "d",
+    "f": "f",
+    "g": "g",
+    "h": "h",
+    "j": "j",
+    "k": "k",
+    "l": "l",
+    "m": "m",
+    "n": "n",
+    "p": "p",
+    "q": "k",
+    "r": "r",
+    "s": "s",
+    "t": "t",
+    "v": "v",
+    "w": "v",
+    "x": "ks",
+    "z": "ts",
+    "þ": "θ",
+    "ð": "ð",
+}
+IPA_class = {
+    "a": a,  # Short vowels
+    "æ": ee,
+    "e": ee,
+    "i": i,
+    "o": o,
+    "ø": oee,
+    "ö": oee,
+    "u": u,
+    "y": y,
+    # Consonants
+    "b": b,
+    "c": k,
+    "d": d,
+    "f": f,
+    "g": g,
+    "h": h,
+    "j": j,
+    "k": k,
+    "l": l,
+    "m": m,
+    "n": n,
+    "p": p,
+    "q": k,
+    "r": r,
+    "s": s,
+    "t": t,
+    "v": v,
+    "w": v,
+    "x": k+s,
+    "z": t+s,
+    "þ": th,
+    "ð": dh,
+}
+
+GEMINATE_CONSONANTS = {
+    "bb": "bː",
+    "dd": "dː",
+    "dh": "ð",
+    "ff": "fː",
+    "gg": "gː",
+    "gh": "ɣ",
+    "kk": "kː",
+    "ll": "lː",
+    "mm": "mː",
+    "nn": "nː",
+    "pp": "pː",
+    "rr": "rː",
+    "ss": "sː",
+    "th": "θ",
+    "tt": "tː",
+    "vv": "vː",
+}
+
+GEMINATE_CONSONANTS_class = {
+    "bb": b.lengthen(),
+    "dd": d.lengthen(),
+    "dh": dh,
+    "ff": f.lengthen(),
+    "gg": g.lengthen(),
+    "gh": gh,
+    "kk": k.lengthen(),
+    "ll": l.lengthen(),
+    "mm": m.lengthen(),
+    "nn": n.lengthen(),
+    "pp": p.lengthen(),
+    "rr": r.lengthen(),
+    "ss": s.lengthen(),
+    "th": th,
+    "tt": t.lengthen(),
+    "vv": v.lengthen(),
+}
+
+DIPHTHONGS_IPA.update(GEMINATE_CONSONANTS)
+DIPHTHONGS_IPA_class.update(GEMINATE_CONSONANTS_class)
+
+# Some Old Norse rules
+# The first rule which matches is retained
+
+rule_th = [Rule(AbstractPosition("inner", [AbstractVowel()], [AbstractVowel()]), th, dh),
+           Rule(AbstractPosition("last", [AbstractConsonant()], None), th, dh),
+           Rule(AbstractPosition("first", None, None), th, th),
+           Rule(AbstractPosition("last", [r.to_abstract()], None), th, dh)]
+
+
+old_swedish_rules = []
+old_swedish_rules.extend(rule_th)
+
+
+if __name__ == "__main__":
```

Comment:
Put this in a test, you don't want calling this every time it runs.

Reply:
Done!